crates/forge-runner/src/gas.rs

```diff
@@ -123,14 +126,46 @@ fn get_messages_costs(
     starknet_gas_usage + sharp_gas_usage
 }
 
+fn clear_compiled_class_hash_update(
+    state_changes: blockifier::state::cached_state::StateChanges,
+) -> blockifier::state::cached_state::StateChanges {
+    // copy-pasted blockifier::state::cached_state::StateChanges
+    struct StateMapsCopy {
+        _nonces: HashMap<ContractAddress, Nonce>,
+        _class_hashes: HashMap<ContractAddress, ClassHash>,
+        _storage: HashMap<StorageEntry, Felt252>,
+        compiled_class_hashes: HashMap<ClassHash, CompiledClassHash>,
+        _declared_contracts: HashMap<ClassHash, bool>,
+    }
+
+    // use to cast blockifier::state::cached_state::StateChanges into same struct but public for us
+    union StateMapsHack {
+        origin: std::mem::ManuallyDrop<blockifier::state::cached_state::StateChanges>,
+        public: std::mem::ManuallyDrop<StateMapsCopy>,
+    }
+
+    let mut u = StateMapsHack {
+        origin: std::mem::ManuallyDrop::new(state_changes),
+    };
+    unsafe {
+        // compiled_class_hash_updates is used only for keeping track of declares
+        // which we don't want to include in gas cost
+        // clear on public struct
+        (*u.public).compiled_class_hashes.clear();
+
+        // cast back into original blockifier::state::cached_state::StateChanges
+        std::mem::ManuallyDrop::<blockifier::state::cached_state::StateChanges>::into_inner(
+            u.origin,
+        )
```

Comment:
Are these comments from the original code?

Reply:
Orginal